datastore/errors/errors.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package errors
+
+import "errors"
+
+var (
+	ErrClosed = errors.New("datastore closed")
+	// ErrConflict is returned when a transaction conflicts with another transaction. This can
+	// happen if the read rows had been updated concurrently by another transaction.
+	//
+	// This error is copied from badger because it is used in a few places outside of the datastore
+	// package. This copy ensures that we can build in environments that do not support badger.
+	//
+	//nolint:revive
```

Comment:
question: What is the `nolint` for?  What error do you get without it?

Reply:
Capital letter at the beginning of the message.

Reply:
This reason nudges me even more towards changing the errors (either in the badger fork, or with a wrapper)

Reply:
Are these code-comments now out of date? These can now be our own errors due to the wrapping?  Can we remove the lint and start them with lower case?

Reply:
Done